### ISSUE TITLE
Some NDCubeSequence methods error when sub-NDCube's data are scalar

### ISSUE THREAD

**Issue by DanRyanIrish** [sunpy/ndcube#78]
Currently if you slice an ```NDCubeSequence``` such that the sub-cubes' data is a scalar rather than an array, an error occurs.  However, this should be a valid slicing operations as an `NDCube` can have a scalar as its data.

Example: Given an ```NDCubeSequence```, ```nds``` of dimensions ```(<Quantity 25 pix>, <Quantity 8 pix.>)```, the following slicing operation raises the following traceback:
```
>>> nds[:, 0]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/sunpy_dev/ndcube/ndcube/mixins/ndslicing.py in __getitem__(self, item)
     29             raise IndexError("None indices not supported")
     30 
---> 31         return super().__getitem__(item)
     32 
     33     def _slice(self, item):

~/anaconda3/envs/irispy-dev/lib/python3.5/site-packages/astropy/nddata/mixins/ndslicing.py in __getitem__(self, item)
     57         # Abort slicing if the data is a single scalar.
     58         if self.data.shape == ():
---> 59             raise TypeError('scalars cannot be sliced.')
     60 
     61         # Let the other methods handle slicing.

TypeError: scalars cannot be sliced.
```

**Comment by Cadair** [comment]
heh, that error looks like it's not in the slicing operation but in the `__repr__`. See what happens when you do:

```python
>>> a = nds[:, 0]
```

**Comment by DanRyanIrish** [comment]
Well spotted!  That doesn't error.  ```NDCubeSequence.dimensions``` also gives an error.  So it's more than just the ```__repr__``` but you're right it doesn't seem to be the slicing per say.

**Comment by DanRyanIrish** [comment]
Fixed by PR #79
